### PR TITLE
need to yield to avoid jumping to finally immediately

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request fixes Bio.SeqIO.GckIO

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/SeqIO/GckIO.py [hunk 1]
```diff
@@ -229,7 +229,8 @@ def GckIterator(source):
             raise ValueError("GCK files must be opened in binary mode.") from None
 
     try:
-        return _parse(handle)
+        records = _parse(handle)
+        yield from records
```
Comment (peterjc) [inline Bio/SeqIO/GckIO.py]: I would have gone for:

```python
yield from _parse(handle)
```

Comment (mdehoon) [reply Bio/SeqIO/GckIO.py]: That is fine too. Feel free to change it.

**Comment by peterjc** [comment]
Did the the problem this fixed show up in the tests? Can you think of a test for it (ideally something general for all of SeqIO).

**Comment by mdehoon** [comment]
It showed up in a test for a different file format under Bio.SeqIO.

> Can you think of a test for it (ideally something general for all of SeqIO).

We could run each test (or some of the tests) on both file names and file handles.

**Comment by peterjc** [comment]
Ah - assuming you are adding the handle/filename flexibility to all the SeqIO parsers, once that's done a new universal test makes sense.

**Pull request merged**